runtime/src/launch/aprun/aprun-utils.c

```diff
@@ -201,6 +202,24 @@ int getCPUsPerCU() {
   return numCPUsPerCU;
 }
 
+const char *getNodeListStr() {
+  return getAprunArgStr(aprun_L);
+}
+char *getNodeListOpt() {
+  const char *nodeList = getenv("CHPL_LAUNCHER_NODELIST");
+  char *nodeListOpt = NULL;
+
+  if (nodeList) {
+    nodeListOpt = chpl_mem_allocMany(strlen(getNodeListStr()) +
```

Comment:
There's also `chpl_mem_alloc()` as a user-facing memory interface function, for when `sizeof(baseType)==1`.  Personally I like that better for cases where either `number` or `size` is 1, because it short-circuits the deep philosophical question "Is this really 1 item of length <len>, or is it <len> items of length 1?"  (The same occurs in the pbs-aprun/ one.)  You don't have to change this on my account, though.

Reply:
Oh, sorry David -- I always forget about `chpl_mem_alloc()` -- feel free to use that if you want